test/new-objectid.test.js

```diff
@@ -0,0 +1,257 @@
+// Copyright IBM Corp. 2013,2020. All Rights Reserved.
+// Node module: loopback-connector-mongodb
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+require('./init.js');
+
+let Book, Chapter;
+const ds = global.getDataSource();
+const objectIDLikeString = '7cd2ad46ffc580ba45d3cb1f';
+const objectIDLikeString2 = '7cd2ad46ffc580ba45d3cb1e';
+const promisify = require('bluebird').promisify;
+
+
+describe.only('New ObjectID', function() {
+
+  const Book = ds.createModel(
+    'Book',
+    {
+      d: {id: true, type: String, mongodb: {dataType: 'objectID'}},
+      oId: {type: String, mongodb: {dataType: 'objectID'}},
+      oIds: {type: [String], mongodb: {dataType: 'objectID'}},
+      title: String,
+      strId: String,
+      strIds: [String],
+      authorId: {type: [String], mongodb: {dataType: 'objectID'}}
+    }
+  );
+
+  const Author = ds.createModel(
+    'Author',
+    {
+      id: {id: true, type: String, mongodb: {dataType: 'objectID'}},
+      name: String
+    }
+  );
+
+  const Deep = ds.createModel(
+    'Deep',
+    {
+      id: {id: true, type: String, mongodb: {dataType: 'objectID'}},
+      innerObj: {
+        innerObj: {
+          innerObj: {
+            innerObj: {
+              sid: String,
+              oid: {type: String, mongodb: {dataType: 'objectID'}}
+            }
+          }
+        }
+      },
+      innerArray: [
+        {
+          sid: String,
+          oid: {type: String, mongodb: {dataType: 'objectID'}},
+          objInsideAnArray: {
+            sid: String,
+            oid: {type: String, mongodb: {dataType: 'objectID'}}
+          },
+          nested: [
+            {
+              sid: String,
+              oid: {type: String, mongodb: {dataType: 'objectID'}}
+            }
+          ]
+        }
+      ]
+    }
+  );
+
+  beforeEach(async () => {
+    await Book.deleteAll();
+    await Author.deleteAll();
+    await Deep.deleteAll();
+
+    Author.hasMany('books');
+    Book.belongsTo('author');
+  });
+
+
+  // SPIKE: in these tests there is no way of knowing if objectID-like strings are actually stored
```

Comment:
For the spike: we test the raw data from database for decimal properties like [this](https://github.com/strongloop/loopback-connector-mongodb/blob/4551cc0428a6954091981eb8e4861e7c1c9b37d8/test/decimal.test.js#L468-L471), hope it helps.

Reply:
Thanks Janny, added.